### DISCUSSION TITLE
Best settings for plotting

### DISCUSSION THREAD

**Discussion by idog8818** [madMAx43v3r/chia-plotter#857] [Q&A]
Hi I have following hardware set, after checking others' plotting time I realized mine is slow and could someone help for settings? 

CPU: E5 * 2, totally 14C28T * 2
RAM: 480G DDR4 2133
OS: Windows
R: 410G Ramdisk, I didn't create 110G and 220G, just set whole and didn't use -2
Z: SSD for temp farm directory for the test 

\chia_plot.exe -n 7 -r 28 -u 128 -t R:\  -d z:\ -c xx -f xx | tee 
Checked CPU usage and can confirm 28Cores are all 90% plus. Here's log.

Final Directory: z:\
Number of Plots: 7
Crafting plot 1 out of 7
Process ID: 5712
Number of Threads: 28
Number of Buckets P1:    2^7 (128)
Number of Buckets P3+P4: 2^7 (128)
Pool Puzzle Hash:  xxxx
Farmer Public Key: xxxx
Working Directory:   R:\
Working Directory 2: R:\
Plot Name: plot-k32-2021-07-21-22-30-58da3d
[P1] Table 1 took 30.5975 sec
[P1] Table 2 took 177.17 sec, found 4294931789 matches
[P1] Table 3 took 241.25 sec, found 4294842132 matches
[P1] Table 4 took 298.079 sec, found 4294620568 matches
[P1] Table 5 took 294.769 sec, found 4294426798 matches
[P1] Table 6 took 283.347 sec, found 4293863558 matches
[P1] Table 7 took 212.451 sec, found 4292671040 matches
Phase 1 took 1537.86 sec
[P2] max_table_size = 4294967296
[P2] Table 7 scan took 16.7239 sec
[P2] Table 7 rewrite took 63.7651 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 26.8364 sec
[P2] Table 6 rewrite took 47.5937 sec, dropped 581446893 entries (13.5413 %)
[P2] Table 5 scan took 25.574 sec
[P2] Table 5 rewrite took 45.4451 sec, dropped 762172616 entries (17.7479 %)
[P2] Table 4 scan took 25.58 sec
[P2] Table 4 rewrite took 44.9279 sec, dropped 828960671 entries (19.3023 %)
[P2] Table 3 scan took 25.5499 sec
[P2] Table 3 rewrite took 44.3513 sec, dropped 855202811 entries (19.9123 %)
[P2] Table 2 scan took 27.1002 sec
[P2] Table 2 rewrite took 44.6576 sec, dropped 865666791 entries (20.1555 %)
Phase 2 took 479.849 sec
Wrote plot header with 252 bytes
[P3-1] Table 2 took 51.0707 sec, wrote 3429264998 right entries
[P3-2] Table 2 took 95.8589 sec, wrote 3429264998 left entries, 3429264998 final
[P3-1] Table 3 took 63.9309 sec, wrote 3439639321 right entries
[P3-2] Table 3 took 89.0736 sec, wrote 3439639321 left entries, 3439639321 final
[P3-1] Table 4 took 63.5158 sec, wrote 3465659897 right entries
[P3-2] Table 4 took 94.7783 sec, wrote 3465659897 left entries, 3465659897 final
[P3-1] Table 5 took 67.5354 sec, wrote 3532254182 right entries
[P3-2] Table 5 took 100.681 sec, wrote 3532254182 left entries, 3532254182 final
[P3-1] Table 6 took 72.442 sec, wrote 3712416665 right entries
[P3-2] Table 6 took 106.969 sec, wrote 3712416665 left entries, 3712416665 final
[P3-1] Table 7 took 63.0777 sec, wrote 4292671040 right entries
[P3-2] Table 7 took 125.495 sec, wrote 4292671040 left entries, 4292671040 final
Phase 3 took 1009.63 sec, wrote 21871906103 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 238.755 sec, final plot size is 108804005194 bytes
Total plot creation time was 3266.27 sec (54.4378 min)
Started copy to z:\plot-k32-2021-07-21-22-30-58da3d

**Comment by gryan315** [comment]
Probably your main issue is using windows, since windows has fairly poor RAM performance, as well as poor performance across NUMA nodes. Your best bet would be to switch to Linux, create 2x 110g tmpfs (possibly even 4) and lock the processes to specific numa nodes with numactl cpunodebind and membind. Most users I've talked to who have used enough RAM to plot entirely in memory saw minimal gain over using NVMe for -t. 

Additionally, what is your RAM configuration to get 480gb? If you're missing a 32gb module you're probably hurting your performance running like that and would likely see improvement by removing the 7 secondary DIMM modules to run with just 256gb.

**Reply by idog8818** [reply]
I'm not familiar with Linux hence tried to avoid of using it. Im using 6*32g and 18*16g ram for totally 480G. 2 physical cpus. It's indeed slow. From the test it only can reach 2000M/s rather than online test of 5000M/s, I tried same test on my DELL PC (16G DDR4) with similar performance. 

Another thing I realized is it will get slow and slow. After 1 day, it will need 80mins per plot. I have to format drive to get better performance.

**Reply by gryan315** [reply]
That is a very bad ram setup and is hurting your performance, you're mixing quad rank 32gb modules with 16gb (most likely) dual rank modules on the same channels. I would recommend just using 16x 16gb modules. On your setup. Also windows is very bad for memory performance, especially when using third party ramdisk software.

**Reply by idog8818** [reply]
I can confirm it's dual rank for all rams..I also checked manual saying if all in. It will run at 1800mhz rather than 2133mhz. All Rams are 2133. I may need to unplug some to test. 

I don't know Linux much.. GUI preferred. Any recommendations on Linux based system for newbie?

**Comment by idog8818** [comment]
Tried uBuntu today and the performance is great..Here's the log. Same server, from 45mins in windows, reduced to 22 mins in uBuntu 

Working Directory:   /mnt/ram/
Working Directory 2: /mnt/ram/
Plot Name: plot-k32-2021-08-05
[P1] Table 1 took 10.7933 sec
[P1] Table 2 took 79.6286 sec, found 4294937704 matches
[P1] Table 3 took 93.6008 sec, found 4294896956 matches
[P1] Table 4 took 103.467 sec, found 4294872337 matches
[P1] Table 5 took 101.96 sec, found 4294714111 matches
[P1] Table 6 took 106.736 sec, found 4294449530 matches
[P1] Table 7 took 85.2667 sec, found 4293875199 matches
Phase 1 took 581.472 sec
[P2] max_table_size = 4294967296
[P2] Table 7 scan took 7.85697 sec
[P2] Table 7 rewrite took 31.7883 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 25.3022 sec
[P2] Table 6 rewrite took 32.372 sec, dropped 581323528 entries (13.5366 %)
[P2] Table 5 scan took 24.4008 sec
[P2] Table 5 rewrite took 30.6071 sec, dropped 761993806 entries (17.7426 %)
[P2] Table 4 scan took 24.0619 sec
[P2] Table 4 rewrite took 31.6708 sec, dropped 828904424 entries (19.2999 %)
[P2] Table 3 scan took 23.3178 sec
[P2] Table 3 rewrite took 30.499 sec, dropped 855116246 entries (19.9101 %)
[P2] Table 2 scan took 23.1783 sec
[P2] Table 2 rewrite took 30.6725 sec, dropped 865607238 entries (20.1541 %)
Phase 2 took 333.386 sec
Wrote plot header with 252 bytes
[P3-1] Table 2 took 27.7357 sec, wrote 3429330466 right entries
[P3-2] Table 2 took 20.1225 sec, wrote 3429330466 left entries, 3429330466 final
[P3-1] Table 3 took 28.4649 sec, wrote 3439780710 right entries
[P3-2] Table 3 took 20.2645 sec, wrote 3439780710 left entries, 3439780710 final
[P3-1] Table 4 took 28.9706 sec, wrote 3465967913 right entries
[P3-2] Table 4 took 20.4485 sec, wrote 3465967913 left entries, 3465967913 final
[P3-1] Table 5 took 29.3025 sec, wrote 3532720305 right entries
[P3-2] Table 5 took 22.1371 sec, wrote 3532720305 left entries, 3532720305 final
[P3-1] Table 6 took 30.4898 sec, wrote 3713126002 right entries
[P3-2] Table 6 took 23.6177 sec, wrote 3713126002 left entries, 3713126002 final
[P3-1] Table 7 took 29.0307 sec, wrote 4293875199 right entries
[P3-2] Table 7 took 27.06 sec, wrote 4293875199 left entries, 4293875199 final
Phase 3 took 312.364 sec, wrote 21874800595 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 66.4346 sec, final plot size is 108821121832 bytes
Total plot creation time was 1293.75 sec (21.5625 min)
